src/toil/test/provisioners/azureProvisionerTest.py

```diff
@@ -0,0 +1,291 @@
+
+log = logging.getLogger(__name__)
+
+
+@needs_azure
+@integrative
+@needs_appliance
+@slow
+class AbstractAzureAutoscaleTest(ToilTest):
+
+    def sshUtil(self, command):
+        baseCommand = ['toil', 'ssh-cluster', '--insecure', '-p=azure', self.clusterName]
+        callCommand = baseCommand + command
+        subprocess.check_call(callCommand)
+
+    def rsyncUtil(self, src, dest):
+        baseCommand = ['toil', 'rsync-cluster', '--insecure', '-p=azure', self.clusterName]
+        callCommand = baseCommand + [src, dest]
+        subprocess.check_call(callCommand)
+
+    def destroyClusterUtil(self):
+        callCommand = ['toil', 'destroy-cluster', '-p=azure', self.clusterName]
+        subprocess.check_call(callCommand)
+
+    def createClusterUtil(self, args=None):
+        if args is None:
+            args = []
+        callCommand = ['toil', 'launch-cluster', self.clusterName, '-p=azure', '--keyPairName=%s' % self.sshKeyName,
+                       '--leaderNodeType=%s' % self.leaderInstanceType, '--zone=%s' % self.azureZone]
+        if self.publicKeyFile:
+            callCommand += ['--publicKeyFile=%s' % self.publicKeyFile]
+        callCommand = callCommand + args if args else callCommand
+        log.info("createClusterUtil: %s" % ' '.join(callCommand))
+        subprocess.check_call(callCommand)
+
+    def cleanJobStoreUtil(self):
+        callCommand = ['toil', 'clean', self.jobStore]
+        subprocess.check_call(callCommand)
+
+    def __init__(self, methodName):
+        super(AbstractAzureAutoscaleTest, self).__init__(methodName=methodName)
+        self.keyName = os.getenv('TOIL_AZURE_KEYNAME')
+        self.publicKeyFile = os.getenv('PUBLIC_KEY_FILE')
+        self.sshKeyName = os.getenv('TOIL_SSH_KEYNAME')
+        self.azureZone = os.getenv('TOIL_AZURE_ZONE')
+
+        self.leaderInstanceType = 'Standard_A2'
+        self.instanceTypes = ["Standard_A3"]
+        self.numWorkers = ['2']
+        self.numSamples = 2
+
+    def setUp(self):
+        super(AbstractAzureAutoscaleTest, self).setUp()
+
+    def tearDown(self):
+        super(AbstractAzureAutoscaleTest, self).tearDown()
+        self.destroyClusterUtil()
+        self.cleanJobStoreUtil()
+
+    def launchCluster(self):
+        self.createClusterUtil()
+
+    @abstractmethod
+    def _getScript(self):
+        """
+        Download the test script needed by the inheriting unit test class.
+        """
+        raise NotImplementedError()
+
+
+    @abstractmethod
+    def _runScript(self, toilOptions):
+        """
+        Modify the provided Toil options to suit the test Toil script, then run the script with
+        those arguments.
+
+        :param toilOptions: List of Toil command line arguments. This list may need to be
+               modified to suit the test script's requirements.
+        """
+        raise NotImplementedError()
+
+    def _test(self, preemptableJobs=False):
```

Comment:
Why was the `assert len(self.getMatchingRoles(self.clusterName)) == 1` taken out?  Shouldn't we have asserts to make sure there is a cluster running?